generate/template/astronomy.ts

```diff
@@ -2418,31 +2373,31 @@ function NormalizeLongitude(lon: number): number {
  *      moment winter begins in the northern hemisphere and summer
  *      begins in the southern hemisphere.
  *
- * @param {(Date | number | Astronomy.AstroTime)} dateStart
+ * @param {(Date | number | AstroTime)} dateStart
```

Comment:
I just realized we can probably replace all of these `Date | number | AstroTime` in the jsdoc comments with `FlexibleDateTime`. I will have to figure out how to update the documentation to explain this to developers who use Astronomy Engine.

Reply:
Typescript users will be able to follow the type definition, but I will run `jsdoc-to-markdown` into a es6 compiled version so we get the `@typedef` included in the markdown file

Reply:
I successfully added a `@typedef` but next to a `constant` because the type doesn't survive to `tsc` :P

Reply:
and also I improved this `@typedef` attaching it to the first function using it, just like you did with `SearchOptions`